grouper/fe/handlers/group_edit_member.py

```diff
@@ -156,3 +134,22 @@ def post(self, *args: Any, **kwargs: Any) -> None:
             )
 
         return self.redirect("/groups/{}?refresh=yes".format(group.name))
+
+    def _get_form(self, my_role: str, member_type: str) -> GroupEditMemberForm:
+        """Get the form with possible role options filled in.
+
+        Owners can edit themselves to downgrade their membership, but not upgrade their
+        membership.  So owners can make themselves members, np-owners, or managers, but np-owners
+        cannot make themselves owners.  The role of a group cannot be changed from member.
+        """
+        form = GroupEditMemberForm(self.request.arguments)
+        form.role.choices = [["member", "Member"]]
+        if my_role in ("owner", "np-owner") and member_type != "group":
+            form.role.choices.append(["manager", "Manager"])
+            form.role.choices.append(["np-owner", "No-Permissions Owner"])
+            if my_role == "owner":
+                form.role.choices.append(["owner", "Owner"])
+        elif member_type == "group" or my_role == "manager":
```

Comment:
Manager -> member is a downgrade, right? Are we not allowing it just because managers shouldn't be able to make any role changes?

Reply:
Yeah, good call, this logic wasn't really correct. I think I've now cleaned it up.